src/vellum/workflows/events/workflow.py

```diff
@@ -203,13 +205,23 @@ def outputs(self) -> OutputsType:
     def final_state(self) -> Optional[StateType]:
         return self.body.final_state
 
-    @field_serializer("body")
-    def serialize_body(
-        self, body: WorkflowExecutionFulfilledBody[OutputsType, StateType], info: SerializationInfo
-    ) -> WorkflowExecutionFulfilledBody[OutputsType, StateType]:
-        return cast(
-            WorkflowExecutionFulfilledBody[OutputsType, StateType], _serialize_body_with_enricher(self, body, info)
-        )
+    @model_serializer(mode="plain", when_used="json")
+    def serialize_model(self, info: SerializationInfo) -> Dict[str, Any]:
+        enriched_body = _serialize_body_with_enricher(self, self.body, info)
+        original_body = self.body
+        self.body = cast(WorkflowExecutionFulfilledBody[OutputsType, StateType], enriched_body)
+        serialized = super().serialize_model(info)
+        self.body = original_body
```

Comment:
DevinAI I'm quite spooked by this approach of reassigning `self.body` attribute during serialization. can you investigate another path? It seems like it should be possible to keep `serialize_body`